csharp/src/Ice/INetworkProxy.cs

```diff
@@ -26,15 +26,15 @@ public interface INetworkProxy
         // Once the connection request has been sent, this is called
         // to prepare and read the response from the proxy server.
         //
-        void BeginRead(Buffer buf);
-        int EndRead(Buffer buf);
+        void BeginRead(ref ArraySegment<byte> buffer);
```

Comment:
The only two implementations of BeginRead are in this file and both allocate a new array, so it sounds like it should be:
`ArraySegment<byte> BeginRead()`

Reply:
Fixed